src/app/pages/courses-page/courses-page.component.html

```diff
@@ -10,6 +10,14 @@
     </div>
   </div>
 
-  <app-course-item *ngFor="let course of courses$ | async" [course]="course" (courseDeleted)="onCourseDelete($event)"></app-course-item>
-  <app-load-more-panel (loadMore)="onLoadMore()"></app-load-more-panel>
-</section>
\ No newline at end of file
+  <ng-container *ngIf="courses$ | async as courses">
+    <ng-container *ngIf="courses.length; else noDataTemplate">
+      <app-course-item *ngFor="let course of courses | orderBy: 'creationDate':'desc'" [course]="course" (courseDeleted)="onCourseDelete($event)"></app-course-item>
+      <app-load-more-panel (loadMore)="onLoadMore()"></app-load-more-panel>
+    </ng-container>
+  </ng-container>
+</section>
+
+<ng-template #noDataTemplate>
+  <app-no-data-message message="No data. Feel free to add new course"></app-no-data-message>
```

Comment:
it seems to be a message block, not `no-data`... there is nothing about data and any check of it existence..

Reply:
Agree